src/test/java/hudson/util/VersionNumberTest.java

```diff
@@ -49,10 +50,12 @@ public void testEarlyAccess() {
        assertTrue(new VersionNumber("2.0.ea1").isNewerThan(new VersionNumber("2.0.ea")));
        assertEquals(new VersionNumber("2.0.ea"), new VersionNumber("2.0.ea0"));
     }
-    
+
+    @Issue("JENKINS-40899")
     public void testSnapshots() {
         assertTrue(new VersionNumber("1.12").isNewerThan(new VersionNumber("1.12-SNAPSHOT (private-08/24/2008 12:13-hudson)")));
         assertTrue(new VersionNumber("1.12-SNAPSHOT (private-08/24/2008 12:13-hudson)").isNewerThan(new VersionNumber("1.11")));
+        assertTrue(new VersionNumber("1.12-SNAPSHOT (private-08/24/2008 12:13-hudson)").isNewerThan(new VersionNumber("1.12-SNAPSHOT")));
```

Comment:
I am not concerned with whether they are equal or not, just that it is considered sufficiently new.